MIGRATING.md

```diff
@@ -26,6 +26,32 @@ The existing `.widgetrc` file used to configure the Widget has been **removed**.
 - }
 ```
 
+## Migrating from 4.x to 5.x
+
+## authClient has been removed
+
+The internal instance of [@okta/okta-auth-js](https://github.com/okta/okta-auth-js) is no longer being exposed. If you need to work with the `okta-auth-js` library, you should add a dependency to [@okta/okta-auth-js](https://github.com/okta/okta-auth-js) and create an instance in your application.
```

Comment:
I don't think we are planning on accepting an instance of authClient from the outside. The internal instance will be customized in certain ways to avoid side-effects. For example, we will forcibly disable autorenew and other "background tasks" to avoid concurrency issues. In the longer term, we intend to break auth-js into smaller pieces and have only the relevant portions used by the SIW.

We could add a bit of an example of sharing config between SIW and auth-js however.

Reply:
Make sense to customized the internal instance but why remove the exposure? It seems convenient if people need to play with some underlining function from auth-js for some cases. (I assume that are probably reasons to expose auth client at first place?)
Otherwise this seems very breaking changes. Developers need to create another auth-js instance which seems redundant.

Reply:
We already have problems with people using an external AuthJS that ends up conflicting with an internal AuthJS - rather than giving devs more ways to complicate matters, we want to reduce them.

AuthJS itself needs to slim down - see the issues we have with the widget when it isn't even using  AuthJS with OIE.  

Why create more ways to have problems?

Reply:
@haishengwu-okta It is "convenient" to have a pre-created auth-js instance, but it has caused a lot of confusion and problems:
- the version which is bundled in SIW is almost never the current version, and if there are bugs, the user cannot install a newer version (because it is bundled in the webpack). It's not even clear what version you are using. Using your own version from NPM gives you control to upgrade it independently of the SIW
- authJS supports a lot of configuration options, passing these all in under `authParams` leads to confusion as to what SIW does and what authJS does. One center of confusion is the TokenManager, which SIW does not (and should not) use.
- we have added more features in authJS that involve it having background logic, such as token auto-renew. Having this logic running in the SIW could cause concurrency issues so we want to forcibly disable these features in the instance used by the SIW. This means we will ignore certain configuration options even if they are passed in authParams. So the instance of auth-js inside SIW is more limited than a standalone instance. We don't really want to document or explain this other than to say the internal instance is internal and we reserve the right to change it as we see fit in the future.
- with IDX all of these things come into play. There is an internal instance that is configured in a certain way for the SIW. We cannot guarantee that you could use this internal instance in the same way as a separate instance. We cannot even guarantee that the internal instance IS auth-js, it may in fact be an instance of idx-js.

Reply:
Moving these changes to a separate PR: https://github.com/okta/okta-signin-widget/pull/1480 we can continue conversation there